db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1161,7 +1161,8 @@ Status DBImpl::CompactFilesImpl(
                            job_context->job_id, version, compaction_job_info);
   }
 
-  if (status.ok()) {
+  IOStatus io_s = compaction_job.io_status();
```

Comment:
I feel we should check for `io_s` inside `CompactionJob::Install()`, and return non-ok status if necessary for `Install()`. Otherwise, it would fee weird if `Install()` returns OK but `io_s` is not.

Reply:
There's a check for `io_s` inside of `CompactionJob::Install()`, but doesn't set the status to non-ok:
https://github.com/facebook/rocksdb/blob/9a63bbd39112a79089533c788fa80465d97d9100/db/compaction/compaction_job.cc#L735-L737
I'm not sure, feels like it doesn't want to fail the compaction by setting the status to non-ok. Is there any non-ok `io_s`, but shouldn't fail the compaction install?

Reply:
Or maybe we should just ignore the `io_s`, the same as memtableFlush:
https://github.com/facebook/rocksdb/blob/25e8c47f83f7e9149535255d015aa33a0d1e7de9/db/db_impl/db_impl_compaction_flush.cc#L240-L263
What do you think?

Reply:
If `versions_->io_status()` is no-ok, then the `InstallCompactionResults()` must have returned non-ok. Otherwise, it seems a bug to me.

Reply:
Make sense. Updated.